src/Microsoft.DotNet.Build.Tasks/PackageFiles/versioning.targets

```diff
@@ -234,6 +236,7 @@
   <Target Name="CreateVersionFileDuringBuild" Condition="'$(SkipVersionGeneration)'!='true' AND '$(ShouldCreateVersionFileDuringBuild)'=='true'" DependsOnTargets="CreateOrUpdateCurrentVersionFile">
     <PropertyGroup>
       <BuildNumberMajor>$(GeneratedBuildNumberMajor)</BuildNumberMajor>
```

Comment:
We can't do that, because this Names will already be set always. They are both set when this file is loaded (versioning.targets) so at the time of the target runs they will always be set already. For servicing, it will be easy enough to just pass in both versions with `msbuild /p:BuildNumberMajor=...` or we could also simply just set `/p:SkipVersionGeneration=true`